### ISSUE TITLE
Slow Response Time on Windows Prompt Compared to WSL

### ISSUE THREAD

**Issue by samer-alhalabi** [ollama/ollama#2699]
When executing prompts on Ollama using Windows version, I experience considerable delays and slowness in response time. However, when running the exact same model and prompt via WSL, the response time is notably faster. Given that the Windows version of Ollama is currently in preview, I understand there may be optimizations underway. Could you provide insight into whether there's a timeline for the next version release that addresses performance ?

**Comment by seanmavley** [comment]
Is the Ollama on windows using GPU? Are you able to confirm that?

Because slower response time on Windows may be because on Windows, somehow Ollama isn't using the GPU, compared to on WSL

**Comment by Huarch** [comment]
Did you run with Windows app? If you run ollama serve through terminal you may find the speed between them should be close. Maybe this is a bug or something.

**Comment by samer-alhalabi** [comment]
> Is the Ollama on windows using GPU? Are you able to confirm that?
> 
> Because slower response time on Windows may be because on Windows, somehow Ollama isn't using the GPU, compared to on WSL

CPU

**Comment by samer-alhalabi** [comment]
> Did you run with Windows app? If you run ollama serve through terminal you may find the speed between them should be close. Maybe this is a bug or something.

Running both through the terminal.

**Comment by dhiltgen** [comment]
We fixed a bug in 0.1.27 where CPU performance on windows was ~1/2 what it should be.  If you're running the prior version, please upgrade and that should fix the performance.  If you are running this version and still see slower performance on native windows vs. WSL2, please attach server logs from both so we can see why it's behaving differently.

**Comment by samer-alhalabi** [comment]
> We fixed a bug in 0.1.27 where CPU performance on windows was ~1/2 what it should be. If you're running the prior version, please upgrade and that should fix the performance. If you are running this version and still see slower performance on native windows vs. WSL2, please attach server logs from both so we can see why it's behaving differently.

Yes, I did notice a noticeable improvement in performance with the newer version [0.1.27].  Thanks for the update.

**Comment by Skyggedans** [comment]
Using deepseek-coder:33b with some simple prompt like "Write a linked list implementation in Rust" mine 4090 has 100% load, but draws only approximately 50 watts from PSU. The request completes in 1-1.5 minutes, which seems way too slow to me.

**Comment by dhiltgen** [comment]
@Skyggedans take a look at #3511 and see if that captures your scenario.  If not, go ahead and file a new issue with your server log so we can see what might be going wrong.

**Comment by shreyasdeodhare** [comment]
Hello @dhiltgen  , I am running the Ollama tinyllama model on the cpu windows machine with Windows10 and I am having the Ollama 0.2.1 version ,but the resposne time for getting the response is about 10 sec which is slow can you please suggest any solution for it I am running the Ollama model through the code
